src/urllib3/util/ssl_.py

```diff
@@ -109,6 +114,9 @@ def assert_fingerprint(cert, fingerprint):
         Fingerprint as string of hexdigits, can be interspersed by colons.
     """
 
+    if cert is None:
```

Comment:
Probably will have to add a test for that.
`cert` can be None because `getpeercert` can return None.

Reply:
`getpeercert` returns `None` only for server SSL sockets, so I added a test that directly tests the `assert_fingerprint` function in `test_ssl.py`